test/unit/managers/ConfigManager.ts

```diff
@@ -157,3 +172,108 @@ test('should have enableOnSession flag', t => {
 
   t.not(fakeMergedConfig.enableOnSession, undefined);
 });
+
+/**
+ * Check Restricted Origin Tests
+ */
+test('should match origins', async t => {
+  const checkRestrictedOriginSpy = sinonSandbox.spy(ConfigHelper, "checkRestrictedOrigin");
+  const doesOriginMatchSpy = sinonSandbox.spy(ConfigHelper, "doesCurrentOriginMatchConfigOrigin");
+  sinonSandbox.stub(OneSignal, "delayedInit");
+  try{
+    await initOneSignalOnUrlWithConfigOrigin("https://example.com", "https://example.com");
+  } catch (e) {
+    // pass
+  }
+
+  t.true(doesOriginMatchSpy.calledOnce);
+  t.true(doesOriginMatchSpy.returnValues[0]);
+  t.false(checkRestrictedOriginSpy.threw());
+});
+
+test('should match origins even with www. variant', async t => {
+  const checkRestrictedOriginSpy = sinonSandbox.spy(ConfigHelper, "checkRestrictedOrigin");
+  const doesOriginMatchSpy = sinonSandbox.spy(ConfigHelper, "doesCurrentOriginMatchConfigOrigin");
+  sinonSandbox.stub(OneSignal, "delayedInit");
+  try{
+    await initOneSignalOnUrlWithConfigOrigin("https://example.com", "https://www.example.com");
+  } catch (e) {
+    // pass
+  }
+
+  t.true(doesOriginMatchSpy.calledOnce);
+  t.true(doesOriginMatchSpy.returnValues[0]);
+  t.false(checkRestrictedOriginSpy.threw());
+});
+
+test('should match origins with both www. and https variants', async t => {
+  const checkRestrictedOriginSpy = sinonSandbox.spy(ConfigHelper, "checkRestrictedOrigin");
+  const doesOriginMatchSpy = sinonSandbox.spy(ConfigHelper, "doesCurrentOriginMatchConfigOrigin");
+  sinonSandbox.stub(OneSignal, "delayedInit");
+  try{
+    // allow https for http-configured sites
+    await initOneSignalOnUrlWithConfigOrigin("https://example.com", "http://www.example.com");
+  } catch (e) {
+    // pass
+  }
+
+  t.true(doesOriginMatchSpy.calledOnce);
+  t.true(doesOriginMatchSpy.returnValues[0]);
+  t.false(checkRestrictedOriginSpy.threw());
+});
+
+test('should throw if current page origin has less strict protocol', async t => {
+  const checkRestrictedOriginSpy = sinonSandbox.spy(ConfigHelper, "checkRestrictedOrigin");
+  const doesOriginMatchSpy = sinonSandbox.spy(ConfigHelper, "doesCurrentOriginMatchConfigOrigin");
+  sinonSandbox.stub(OneSignal, "delayedInit");
+  try{
+    // don't allow http for https-configured sites
+    await initOneSignalOnUrlWithConfigOrigin("http://example.com", "https://www.example.com");
+  } catch (e) {
+    // pass
+  }
+
+  t.true(doesOriginMatchSpy.calledOnce);
+  t.false(doesOriginMatchSpy.returnValues[0]);
+  t.true(checkRestrictedOriginSpy.threw());
+});
+
+test('should throw if origins do not match', async t => {
+  const checkRestrictedOriginSpy = sinonSandbox.spy(ConfigHelper, "checkRestrictedOrigin");
+  const doesOriginMatchSpy = sinonSandbox.spy(ConfigHelper, "doesCurrentOriginMatchConfigOrigin");
+  sinonSandbox.stub(OneSignal, "delayedInit");
+  try{
+    await initOneSignalOnUrlWithConfigOrigin("https://www.example.com", "https://example.com");
```

Comment:
This is safe since host is the same for both and they are both HTTPS. However not sure if you can subscribe and / or have two different service workers with and without `www.`. After we figure this out we will know if we should support this or not a make a note here.